src/pkg/util/fs/mount/mount.go

```diff
@@ -0,0 +1,370 @@
+		setContext = authorizedFS[fstype].context
+	}
+	if setContext {
+		context := fmt.Sprintf("context=%s", p.Context)
+		mountOptions = append(mountOptions, context)
+	}
+	p.points = append(p.points, specs.Mount{
+		Source:      source,
+		Destination: dest,
+		Type:        fstype,
+		Options:     mountOptions,
+	})
+	return nil
+}
+
+// GetAll returns all registered mount points
+func (p *Points) GetAll() []specs.Mount {
+	return p.points
+}
+
+// Get returns registered mount points with the matched destination
+func (p *Points) Get(dest string) []specs.Mount {
+	mounts := []specs.Mount{}
+	for _, point := range p.points {
+		if point.Destination == dest {
+			mounts = append(mounts, point)
+			break
+		}
+	}
+	return mounts
+}
+
+// Remove removes mount points identified by destination
+func (p *Points) Remove(dest string) {
+	for i := len(p.points) - 1; i >= 0; i-- {
+		if p.points[i].Destination == dest {
+			p.points = append(p.points[:i], p.points[i+1:]...)
+		}
+	}
+}
+
+// RemoveAll removes all mounts points from list
+func (p *Points) RemoveAll() {
+	p.points = nil
+}
+
+// Import imports a mount point list
+func (p *Points) Import(points []specs.Mount) error {
+	for _, point := range points {
+		var err error
+		var offset uint64
+
+		flags, options := ConvertOptions(point.Options)
+		// check if this is a mount point to remount
+		if flags&syscall.MS_REMOUNT != 0 {
+			if err = p.AddRemount(point.Destination, flags); err == nil {
+				continue
+			}
+		}
+		// check if this is a bind mount point
+		if flags&syscall.MS_BIND != 0 {
+			if err = p.AddBind(point.Source, point.Destination, flags); err == nil {
+				continue
+			}
+		}
+		// check if this is an image mount point
+		for _, option := range options {
+			if strings.HasPrefix(option, "offset=") {
+				fmt.Sscanf(option, "offset=%d", &offset)
+				break
+			}
+		}
+		if err = p.AddImage(point.Source, point.Destination, point.Type, flags, offset); err == nil {
+			continue
+		}
+		// check if this is a filesystem or overlay mount point
+		if point.Type != "overlay" {
+			if err = p.AddFS(point.Destination, point.Type, flags, strings.Join(options, ",")); err == nil {
+				continue
+			}
+		} else {
+			lowerdir := ""
+			upperdir := ""
+			workdir := ""
+			for _, option := range options {
+				if strings.HasPrefix(option, "lowerdir=") {
+					fmt.Sscanf(option, "lowerdir=%s", &lowerdir)
+				} else if strings.HasPrefix(option, "upperdir=") {
+					fmt.Sscanf(option, "upperdir=%s", &upperdir)
+				} else if strings.HasPrefix(option, "workdir=") {
+					fmt.Sscanf(option, "workdir=%s", &workdir)
+				}
+			}
+			if err = p.AddOverlay(point.Destination, flags, lowerdir, upperdir, workdir); err == nil {
+				continue
+			}
+		}
+		p.RemoveAll()
+		return fmt.Errorf("can't import mount points list: %s", err)
+	}
+	return nil
+}
+
+// AddImage adds an image mount point
+func (p *Points) AddImage(source string, dest string, fstype string, flags uintptr, offset uint64) error {
```

Comment:
Can we make this function additionally take `length uint64`. This way we can natively mount a partition from SIF

Reply:
There is offset argument ;)